code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm

```diff
@@ -494,6 +511,26 @@ GLOBAL_LIST_EMPTY_TYPED(air_alarms, /obj/machinery/airalarm)
 
 	if(danger_level)
 		alarm_manager.send_alarm(ALARM_ATMOS)
+		if(pressure <= WARNING_LOW_PRESSURE && temp <= BODYTEMP_COLD_WARNING_1+10)
+			warning_message = "Danger! Low pressure and temperature detected."
+			return
+		if(pressure <= WARNING_HIGH_PRESSURE && temp >= BODYTEMP_HEAT_WARNING_1-27)
+			warning_message = "Danger! High pressure and temperature detected."
+			return
+		if(pressure <= WARNING_LOW_PRESSURE)
+			warning_message = "Danger! Low pressure detected."
+			return
+		if(pressure >= WARNING_HIGH_PRESSURE)
+			warning_message = "Danger! High pressure detected."
+			return
+		if(temp <= BODYTEMP_COLD_WARNING_1+10)
+			warning_message = "Danger! Low temperature detected."
+			return
+		if(temp >= BODYTEMP_HEAT_WARNING_1-27)
+			warning_message = "Danger! High temperature detected."
+			return
+		warning_message = "Danger! Atmospheric issue detected."
```

Comment:
You should change all of this to a pure `get_warning_message()` proc, which would turn this into `warning_message = get_warning_message()`, and turn all these into just `return "message here"`

Reply:
I just spend an hour fiddling around with trying to make this suggestion work and kept running into weird `Exception has occurred: Cannot execute null.total moles().` when trying to get the same data on pressure and temperature that `check_danger()` uses for the original implementation I did.

Even if I figure out the issue, `get_warning_message()` still needs a huge copy paste of what is inside of `check_danger()` to properly see the pressure and temperature of the tile so it wouldn't even make the code cleaner.

Unless I totally misunderstood what you want the new proc to be.

Reply:
That doesn't make any sense, I don't know why you would face any issues with this.

What I want is for this code to be in its own proc, and `return` instead of setting the variable. The only code it would need to execute on its own is the code needed to get `pressure` and `temp`. That's it.

Reply:
![pressure error](https://github.com/tgstation/tgstation/assets/55374212/0f9c9ec0-5667-4a9b-aa16-3d390f814bfe)

I don't know why the error pops. This is with minimal copy pasta from `check_danger()` but even if I add all of it, the error still shows here, but not on the original proc.

Reply:
You aren't passing in any parameters then

Reply:
I spent another 2 hours trying to move it into another proc running on other new issues while still having to copy paste more vars into this new proc to even work.

So I give up, I dislike the copy pasta more than the if chain on the original proc and I'm not spending more hours trying to work on this, I'm already lost on what I attempted, what failed, and whatever else after this long. Feel free to close the PR if a new proc is a requirement in your opinion.